net/adguardhome/Makefile

```diff
@@ -0,0 +1,67 @@
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE.txt for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=adguardhome
+PKG_VERSION:=0.102.0
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_VERSION:=v$(PKG_VERSION)
+PKG_SOURCE_URL:=https://github.com/AdguardTeam/AdGuardHome
+PKG_MIRROR_HASH:=c457088177907a032d67c67ce2136bbc8e462be4cf61cd4274424b5e88dfa499
+
+PKG_LICENSE:=GPL-3.0-only
+PKG_LICENSE_FILES:=LICENSE.txt
+PKG_MAINTAINER:=Dobroslaw Kijowski <dobo90@gmail.com>
+
+PKG_BUILD_DEPENDS:=golang/host node/host
+PKG_BUILD_PARALLEL:=1
+PKG_USE_MIPS16:=0
+
+GO_PKG:=github.com/AdguardTeam/AdGuardHome
+GO_PKG_EXCLUDES:=dhcpd/standalone
+GO_PKG_LDFLAGS:=-s -w
+GO_PKG_LDFLAGS_X:=main.version=$(PKG_VERSION) main.channel=release
+
+include $(INCLUDE_DIR)/package.mk
+include ../../lang/golang/golang-package.mk
+
+define Package/adguardhome
+  SECTION:=net
+  CATEGORY:=Network
+  TITLE:=Network-wide ads and trackers blocking DNS server
+  URL:=https://github.com/AdguardTeam/AdGuardHome
+  DEPENDS:=$(GO_ARCH_DEPENDS) +ca-bundle
+endef
+
+define Package/adguardhome/conffiles
+/etc/adguardhome.yaml
+endef
+
+define Package/adguardhome/description
+Free and open source, powerful network-wide ads and trackers blocking DNS server.
+endef
+
+define Build/Compile
+( \
+  cd $(PKG_BUILD_DIR) ; \
+  npm --prefix client ci ; \
+  npm --prefix client run build-prod ; \
+  GOOS=$$$$(go env GOOS) GOARCH=$$$$(go env GOARCH) GO111MODULE=off go get -v github.com/gobuffalo/packr/... ; \
```

Comment:
I would prefer adding a separate package (with a known version/commit) for packr and using it as a host tool instead of downloading and building it here.

Reply:
GOARM isn't set anywhere in Makefile so I've skipped it.

Reply:
I've tried to do it but I'm sure that I haven't done it properly (you can check it here https://gist.github.com/dobo90/4c0a78dff4bd30358f38aa5cd95727b8). I haven't found any host package using golang. So I've copy paste a lot of variables form golang-package.mk and golang-values.mk. Any suggestion how to make it look good are welcome.

On the other side: I saw that recent version of AdGuardHome has introduced usage of go generate. I've updated the version of the software and made Makefile use the go generate. There was an issue that it was compiling packr for the target architecture but I've pushed a separate commit to address this.
I've made it download specified version of packr. Does this meet your requirements?

Reply:
I can see packr here would be pinned to a known version, but I'm still not crazy about (re-)building a host tool every time this package is built.

I took a brief look at your packr Makefile gist - thanks for attempting it. I think I thought host Go packages were already possible but obviously I was wrong :joy:

I think going forward, there may be more host tools written in Go. Let me think about how it can be done in a similar manner as target Go packages (probably with something like a golang-host.mk include file).

Reply:
I've added support for building host Go programs in #13570. Once that is merged, I will open another PR to add host packr, then this package can add `packr/host` to `PKG_BUILD_DEPENDS`. You can either patch out the `go:generate go install` line and call `go generate` or just call `packr -z` directly during `Build/Configure`.